### PR TITLE
shipit_static_analysis: Import libraries to publish reviews to MozReview.

### PR DESCRIPTION
**Pull request by jankeromnes**
This should get us started with #364.

### CODE REVIEW

**Comment by jankeromnes** [comment]
@La0 I simply imported [batchreview.py](https://hg.mozilla.org/hgcustom/version-control-tools/file/tip/mozreviewpulse/mozreviewpulse/batchreview.py) into our repository. Where should I hack Nix's configuration in order to make the imports work? (Mostly ReviewBoard's `rbtools` module, but I guess `json` and `logging` will also need to work, unless we tweak the file to our preference.)

File: src/shipit_static_analysis/requirements.nix [hunk 1]
```diff
@@ -91,6 +91,24 @@ let
...
 
+    "RBTools" = python.mkDerivation {
+      name = "RBTools-0.7.10";
+      src = pkgs.fetchurl { url = "https://pypi.python.org/packages/b3/9b/f2824009aefc5876a10d122424f83d31572ead24aaec869837a86dbd7156/RBTools-0.7.10.tar.gz"; sha256 = "7da97baa0dd447b3d83e1eed8ff41eee32d9195925ac08dc2c7b019b9b85ae8d"; };
+      doCheck = commonDoCheck;
+      buildInputs = commonBuildInputs;
+      propagatedBuildInputs = [
+      self."six"
+      self."tqdm"
+    ];
```
Comment (jankeromnes) [inline src/shipit_static_analysis/requirements.nix]: Nit: Indentation problem on the 3 lines above.

Comment (La0) [inline src/shipit_static_analysis/requirements.nix]: That code is generated...

**Comment by jankeromnes** [comment]
@garbas This pull request implements the first step toward exporting our detected code defects to MozReview (drive-by reviews submitted directly by our bot, see #364).

Could you please have a look?

File: src/shipit_static_analysis/requirements.nix (hunk 1 above)
Comment (garbas) [inline src/shipit_static_analysis/requirements.nix]: i guess we would have to fix this here https://github.com/garbas/pypi2nix/blob/master/src/pypi2nix/stage3.py#L179

File: src/shipit_static_analysis/shipit_static_analysis/batchreview.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import json
+import logging
```
Comment (garbas) [inline src/shipit_static_analysis/shipit_static_analysis/batchreview.py]: use this snippet to get a logger

```python
import cli_common.log
logger = cli_common.log.get_logger(__name__)
```

File: src/shipit_static_analysis/shipit_static_analysis/batchreview.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        The ``logger`` object is a logger to be used.
+        '''
+
+        self.api_root = api_root
+        self.review_request_id = review_request_id
+        self.diff_revision = diff_revision
+        self.max_comments = max_comments
+        self.comments = []
+
+        if logger is None:
```
Comment (garbas) [inline src/shipit_static_analysis/shipit_static_analysis/batchreview.py]: have logger be a global of this module, as done here -> https://github.com/mozilla-releng/services/blob/master/src/shipit_uplift/shipit_uplift/api.py#L27

**Comment by jankeromnes** [comment]
Thanks for the suggestion! I refactored `logger` to be a `cli_common.log` based module global. Please have another look.

**Pull request merged**